### PR TITLE
Remove user-excluded triggers from trigger-groups on filter panel

### PR DESCRIPTION
**Pull request by tausifpatel**
When the triggers are excluded for a logged in user, then those excluded triggers should not appear in the trigger groups on filter panel - making filter panel clutter free. 

**Before triggers excluded**
Screenshot 2024-11-11 at 4 44 55 PM

**After trigger TRPR0001 is excluded for user**
Screenshot 2024-11-11 at 4 45 07 PM

### CODE REVIEW

**Review by richrace — DISMISSED** [review]

File: packages/ui/cypress/component/CaseListFilter/TriggerGroups.cy.tsx [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
   const dispatch = () => {}
+  const currentUser = { excludedTriggers: [] } as unknown as DisplayFullUser
 
   it("contains Bails checkbox", () => {
-    cy.mount(<TriggerGroups dispatch={dispatch} reasonCodes={[]} />)
+    cy.mount(
+      <CurrentUserContext.Provider value={{ currentUser }}>
+        <TriggerGroups dispatch={dispatch} reasonCodes={[]} />
+      </CurrentUserContext.Provider>
+    )
```
Comment (richrace) [inline packages/ui/cypress/component/CaseListFilter/TriggerGroups.cy.tsx:14]: You can move this into `beforeEach`

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      cy.loginAs("SupervisorWithExcludedTriggers")
+      cy.visit("/bichard")
+
+      cy.get("input#bails").should("exist")
+      cy.get("input#bails").click()
+
+      cy.get(".govuk-checkboxes__item").contains("PR08 - Breach of bail").should("not.exist")
+      cy.get(".govuk-checkboxes__item").contains("PR10 - Conditional bail").should("exist")
+      cy.get(".govuk-checkboxes__item").contains("PR20 - Breach").should("exist")
+      cy.get(".govuk-checkboxes__item").contains("PR30 - Pre-charge bail application").should("exist")
```
Comment (richrace) [inline packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:12]: You can be more explicit with the finders:

```typescript
cy.get(".govuk-checkboxes__item #trpr0008 + label").should("not.exist")
cy.get(".govuk-checkboxes__item #trpr0010 + label").contains("PR10 - Conditional bail").should("exist")
cy.get(".govuk-checkboxes__item #trpr0020 + label").contains("PR20 - Breach").should("exist")
cy.get(".govuk-checkboxes__item #trpr0030 + label").contains("PR30 - Pre-charge bail application").should("exist")
```

(Remove the SonarLint duplication warning)

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+describe("Triggers excluded at user level", () => {
+    it("should not show excluded triggers in trigger groups and reason codes in the reason codes input when we click on Bails", () => {
```
Comment (richrace) [inline packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:2]: Nitpick: 2 spaces not 4 spaces

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      cy.loginAs("SupervisorWithExcludedTriggers")
+      cy.visit("/bichard")
```
[the remark this answers is not in the document]
Comment (tausifpatel) [reply packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:4]: Thanks for suggestion, added in `beforeEach` block

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts (hunk 2 above)
Comment (tausifpatel) [reply packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:12]: Thanks for suggesting, applied changes to remove sonar warnings

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts (hunk 3 above)
Comment (tausifpatel) [reply packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:2]: Adjusted the spacing

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  it("should not show excluded triggers in trigger groups and reason codes in the reason codes input when we click on Warrants", () => {
+    cy.get("input#warrants").should("exist")
+    cy.get("input#warrants").click()
+
+    cy.get(".govuk-checkboxes__item #trpr0002 + label").should("not.exist")
+    cy.get(".govuk-checkboxes__item #trpr0012 + label").contains("PR12 - Warrant withdrawn").should("exist")
+
+    cy.get("input#reasonCodes").should("value", "PR12")
+  })
+})
```
Comment (richrace) [inline packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:54]: Nitpick: Add a new line at the end of the file

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+describe("Triggers excluded at user level", () => {
+  beforeEach(() => {
+    cy.loginAs("SupervisorWithExcludedTriggers")
+    cy.visit("/bichard")
+  })
+  it("should not show excluded triggers in trigger groups and reason codes in the reason codes input when we click on Bails", () => {
```
[the remark this answers is not in the document]
Comment (tausifpatel) [reply packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:6]: Added new line :)

File: packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts (hunk 5 above)
Comment (tausifpatel) [reply packages/ui/cypress/e2e/case-list/filter-cases/triggersExcludedAtUserLevel.cy.ts:54]: Added new line :)

**Review by richrace — APPROVED** [review]

**Pull request merged**